src/Forms/ImportPreviewForm/GroupPreview/GroupPreview.jsx

```diff
@@ -0,0 +1,244 @@
+import React, { useState, useCallback, useEffect, useMemo } from "react";
+import { useTranslation } from "react-i18next";
+
+import CheckCircleIcon from "@material-ui/icons/CheckCircle";
+
+import Button from "../../../Components/Button/Button.jsx";
+import Details from "../../../Components/Details/Details.jsx";
+import Select, {
+  SelectOptionWithFlag,
+} from "../../../Components/Form/Select/Select.jsx";
+import Switch from "../../../Components/Form/Switch/Switch.jsx";
+import TextInput from "../../../Components/Form/TextInput/TextInput.jsx";
+import Modal from "../../../Components/Modal/Modal.jsx";
+import Table from "../../../Components/Table/Table.jsx";
+import PackageForm from "../../../Forms/PackageForm/PackageForm.jsx";
+
+import useSnack from "../../../hooks/useSnack.js";
+import { getPackages, importVocabs } from "../../../utils/api.js";
+import { delay } from "../../../utils/index.js";
+
+import "./GroupPreview.scss";
+
+const GroupPreview = ({ defaultPackage, onSubmitCallback, importedData }) => {
+  const { t } = useTranslation();
+  const { showSnack } = useSnack();
+
+  const [importedGroup, setImportedGroup] = useState(importedData);
+  const [packages, setPackages] = useState([]);
+  const [packageItems, setPackageItems] = useState([]);
+  const [selectedPackage, setSelectedPackage] = useState(defaultPackage);
+  const [showAddPackage, setShowAddPackage] = useState(false);
+  const [activateVocabs, setActivateVocabs] = useState(false);
+  const [vocabsActive, setVocabsActive] = useState(true);
+
+  const openPackageModal = useCallback(() => {
+    setShowAddPackage(true);
+  }, []);
+
+  const closePackageModal = useCallback(() => {
+    setShowAddPackage(false);
+  }, []);
+
+  const fetchPackages = useCallback(() => {
+    getPackages(true)
+      .then(({ data }) => {
+        setPackages(data);
+      })
+      .catch((err) => {
+        showSnack("error", t("global.fetchError"));
+      });
+  }, [showSnack, t]);
+
+  useEffect(() => {
+    fetchPackages();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const packageAdded = useCallback(
+    (newPackage) => {
+      setSelectedPackage({
+        value: newPackage.id,
+        label: (
+          <SelectOptionWithFlag
+            name={newPackage.name}
+            foreignLanguageCode={newPackage.foreignWordLanguage}
+            translatedLanguageCode={newPackage.translatedWordLanguage}
+          />
+        ),
+      });
+      closePackageModal();
+      fetchPackages();
+    },
+    [closePackageModal, fetchPackages]
+  );
+
+  const submitImport = () => {
+    importVocabs({
+      data: importedGroup,
+      languagePackageId: selectedPackage.value,
+      active: true,
+      activate: false,
+    })
+      .then(async (response) => {
+        showSnack("success", t("screens.allGroups.importSuccessMessage"));
+        await delay(1000);
```

Comment:
What is the intent behind this delay?

Reply:
After the group was successfully imported, ``onSubmitCallback`` is executed, which reloads the page. This reload is removing the showSnack information, so the delay will stop the process for 1 second in order to show the snack information, then reloading the page.